lms/djangoapps/courseware/tests/test_inline_analytics_integration.py

```diff
@@ -303,70 +301,182 @@ def test_process_analytics_answer_dist_radio(self):
+                'correct': True,
+                'count': 23,
+                'value_id': 'choice_1',
+                'answer_value_text': 'Option 2',
+                'answer_value_numeric': 'null',
+                'variant': None,
+                'created': '2014-10-15T101351',
             },
         ]
 
         question_types_by_part = {
-            "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": "checkbox",
+            'i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1': 'checkbox',
         }
 
         num_options_by_part = {
-            "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": 0,
+            'i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1': 0,
         }
 
         processed_data = {
-            "count_by_part": {
+            'count_by_part': {
             },
-            "data_by_part": {
+            'data_by_part': {
             },
-            "message_by_part": {
-                "i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1": "The analytics cannot be displayed for this question as the number of rows returned did not match the question definition."
+            'message_by_part': {
+                'i4x-A-B-problem-f3ed0ba7f89445ee9a83541e1fc8a2f2_2_1': (
+                    'The analytics cannot be displayed for this question as the number of rows returned did '
+                    'not match the question definition.'
+                )
             },
-            "last_update_date": "Oct 15, 2014 at 10:13 UTC"
+            'last_update_date': 'Oct 15, 2014 at 10:13 UTC'
         }
 
         return_json = process_analytics_answer_dist(data, question_types_by_part, num_options_by_part)
         self.assertEquals(json.loads(return_json.content), processed_data)
+
+
+@override_settings(ANALYTICS_DATA_URL='dummy_url',
+                   ZENDESK_URL=ZENDESK_URL)
+class InlineAnalyticsAnswerDistributionWithOverrides(ModuleStoreTestCase):
+
+    def setUp(self):
+        super(InlineAnalyticsAnswerDistributionWithOverrides, self).setUp()
+        self.user = UserFactory.create()
+        self.factory = RequestFactory()
+        self.course = CourseFactory.create(
+            org='A',
+            number='B',
+            display_name='C',
+        )
+        self.staff = StaffFactory(course_key=self.course.id)
+        self.instructor = InstructorFactory(course_key=self.course.id)
+
+        analytics_data = {
+            'module_id': '123',
+            'question_types_by_part': 'radio',
+            'num_options_by_part': 6,
+            'course_id': 'A/B/C',
+        }
+        json_analytics_data = json.dumps(analytics_data)
+        self.data = {'data': json_analytics_data}
+        self.zendesk_response = (
+            'A problem has occurred retrieving the data, to report the problem click '
+            '<a href="{ZENDESK_URL}/hc/en-us/requests/new">here</a>'
+        ).format(ZENDESK_URL=ZENDESK_URL)
+
+    def test_regular_user(self):
+
```

Comment:
Nit: this and the next six functions begin with an empty line, which doesn't agree with PEP-8.